### ISSUE TITLE
[class.access.base] p5 Whether the base-specifier-list of a derived class is a point R where protected member is accessible?

### ISSUE THREAD

**Issue by xmh0511** [cplusplus/draft#5404]
We define whether a member of naming class `N` is accessible at R is as the following: 
> A member m is accessible at the point R when named in class N if
>> - [...]
>> - m as a member of N is protected, ..., or **in a member of a class P derived from N**, where m as a member of P is public, private, or protected, or

It does not mention the *base-specifier-list* of class P. Consider this example:
````cpp
struct N{
 protected:
    struct Base{};
};
struct P: N::Base, N{};
````
member `Base` is named in class `N` and `N` is a base class of class `P`. However, as defined above, we just say `Base` is accessible in the member of `P`. We do not explicitly say that `Base` is accessible at R  that occurs in * base-specifier-list* of `P`. Although, we mention it in [class.access.general] p1
> A member of a class can be
>> - [...]
>> - protected, that is, it can be named only by members and friends of the class in which it is declared, by classes derived from that class, and by their friends (see [class.protected]); or 

However, it is not the definition of whether the member is **accessible**.

**Comment by jensmaurer** [comment]
CWG372

**Comment by jensmaurer** [comment]
This is the example in 11.8.1 p7 `struct D: A::B, A { };`

**Comment by xmh0511** [comment]
> This is the example in 11.8.1 p7 `struct D: A::B, A { };`

Yes, we just lack a wording that can interpret why it's well-formed. I think it's hard to simply say that `R` occurs in *base-specifier-list* of class `P`. 

````cpp
template<class T>
struct E: T::Base, T{};
E<N> e;
````
`T::Base` is a dependent name, which should be checked in the instantiation context, we have no *base-specifier-list* on grammar at that context.

**Comment by jensmaurer** [comment]
I think we have plenty of precedence in the standard where we refer to grammar, but need to await instantiation for the rule to make sense.

**Comment by xmh0511** [comment]
> but need to await instantiation for the rule to make sense.

The concern is that in the implicit instantiation of `E<N>`, we do not have grammar stuff that looks like
````cpp
struct E<N>: N::Base, N{};
````
Where the stuff `N::Base, N` that follows `:` is the actual syntactically *base-specifier-list*(on grammar). This is similar to that: the implicitly function call is not an expression of the grammar, we have no way to express the grammar that would appear in the instantiation stuff
